src/hooks/cakeVault/useGetVaultSharesInfo.ts

```diff
@@ -0,0 +1,33 @@
+import { useEffect, useState } from 'react'
+import BigNumber from 'bignumber.js'
+import { convertSharesToCake } from 'views/Pools/helpers'
+import { useCakeVaultContract } from 'hooks/useContract'
+
+const useGetVaultSharesInfo = (lastUpdated?: number) => {
+  const cakeVaultContract = useCakeVaultContract()
+  const [totalShares, setTotalShares] = useState(null)
+  const [totalCakeInVault, setTotalCakeInVault] = useState(null)
+  const [pricePerFullShare, setPricePerFullShare] = useState(null)
+
+  useEffect(() => {
+    const getPricePerShare = async () => {
+      const sharePrice = await cakeVaultContract.methods.getPricePerFullShare().call()
+      setPricePerFullShare(new BigNumber(sharePrice))
+    }
+    getPricePerShare()
+  }, [cakeVaultContract, lastUpdated])
+
+  useEffect(() => {
+    const getTotalShares = async () => {
+      const shares = await cakeVaultContract.methods.totalShares().call()
+      const { cakeAsBigNumber } = convertSharesToCake(new BigNumber(shares), pricePerFullShare)
+      setTotalShares(new BigNumber(shares))
+      setTotalCakeInVault(cakeAsBigNumber)
+    }
+    getTotalShares()
+  }, [cakeVaultContract, lastUpdated, pricePerFullShare])
```

Comment:
Why are these separate effects?

Reply:
Sometimes, Hachioji, even I don't know why I do certain things.

Reply:
Oh sorry - not, there is logic here. The second effect is reliant on the thing that the first effect fetches 

`pricePerFullShare` (first effect) is necessary to figure out the `totalCakeInVault` (second effect)

Reply:
Batched requests and blended the effects